osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -178,6 +188,71 @@ public void Import(BeatmapSetInfo beatmapSetInfo)
         }
 
         /// <summary>
+        /// Downloads a beatmap.
+        /// </summary>
+        /// <param name="beatmapSetInfo">The <see cref="BeatmapSetInfo"/> to be downloaded.</param>
+        /// <returns>The new <see cref="DownloadBeatmapSetRequest"/>, or null if a download already exists for the same beatmap.</returns>

```

Comment:
not sure, but it may make more sense to return the existing download if one exists.

Reply:
i'm not sure either I went back and forth on it and finally decided on null to explicitly disallow any chance of multiple binding download event handlers in download creators (in this case `DirectPanel`). maybe renaming this method to `CreateDownload` will make the null sensible?